exchanges/gateio/gateio_ws_option.go

```diff
@@ -558,83 +545,68 @@ func (g *Gateio) processOptionsCandlestickPushData(data []byte) error {
 	return nil
 }
 
-func (g *Gateio) processOrderbookTickerPushData(data []byte) error {
-	var response WsResponse
-	orderbookTicker := WsOptionsOrderbookTicker{}
-	response.Result = &orderbookTicker
-	err := json.Unmarshal(data, &orderbookTicker)
+func (g *Gateio) processOrderbookTickerPushData(incoming []byte) error {
+	var data WsOptionsOrderbookTicker
+	err := json.Unmarshal(incoming, &data)
 	if err != nil {
 		return err
 	}
-	g.Websocket.DataHandler <- &response
+	g.Websocket.DataHandler <- &data
 	return nil
 }
 
-func (g *Gateio) processOptionsOrderbookSnapshotPushData(event string, data []byte) error {
+func (g *Gateio) processOptionsOrderbookSnapshotPushData(event string, incoming []byte, pushTime int64) error {
 	if event == "all" {
-		var response WsResponse
-		snapshot := WsOptionsOrderbookSnapshot{}
-		response.Result = &snapshot
-		err := json.Unmarshal(data, &response)
+		var data WsOptionsOrderbookSnapshot
+		err := json.Unmarshal(incoming, &data)
 		if err != nil {
 			return err
 		}
-		pair, err := currency.NewPairFromString(snapshot.Contract)
+		pair, err := currency.NewPairFromString(data.Contract)
```

Comment:
Optional: Could set Contract to type currency.Pair and have it Unmarshal it for you as empty pair checks happen later down the line in orderbook processing